src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -345,6 +348,27 @@ def post_process(self, r):
 
 class ContainerappEnvPreviewCreateDecorator(ContainerAppEnvCreateDecorator):
 
+    def construct_payload(self):
+        super().construct_payload()
+
+        self.set_up_infrastructure_resource_group()
+
+    def validate_arguments(self):
+        super().validate_arguments()
+
+        # Infrastructure Resource Group
+        if self.get_argument_infrastructure_resource_group() is not None:
+            if not self.get_argument_infrastructure_subnet_resource_id():
+                raise RequiredArgumentMissingError("Cannot use --infrastructure-resource-group/-i without "
+                                                   "--infrastructure-subnet-resource-id/-s")
+            if not self.get_argument_enable_workload_profiles():
+                raise RequiredArgumentMissingError("Cannot use --infrastructure-resource-group/-i without "
+                                                   "--enable-workload-profiles/-w")
+
+    def set_up_infrastructure_resource_group(self):
+        if self.get_argument_enable_workload_profiles() and self.get_argument_infrastructure_subnet_resource_id() is not None:
+            self.managed_env_def["properties"]["InfrastructureResourceGroup"] = self.get_argument_infrastructure_resource_group()
```

Comment:
As you described in argument, a random rg name will be generated. It is done in RP?